### PR TITLE
Lint print!("...\n") (closes #455)

### LINKED ISSUE #455: print!("..\n") instead of println!("..")
The latter is more readable and will do the right thing on all systems, whereas the former only works on UNIX, IIRC.

The hard thing is the macro expansion.

### PR DESCRIPTION
**Pull request by birkenfeld**

### CODE REVIEW

**Comment by birkenfeld** [comment]
Not sure what to do about the travis failure, looks unrelated.

**Comment by birkenfeld** [comment]
Ah. https://internals.rust-lang.org/t/cargo-regression-regex-crate-broken-on-2016-08-05-nightly/3819

**Comment by oli-obk** [comment]
needs a rebase

**Comment by birkenfeld** [comment]
done!

**Comment by oli-obk** [comment]
lgtm

File: clippy_lints/src/format.rs [hunk 1]
```diff
@@ -69,11 +69,10 @@ impl LateLintPass for Pass {
     }
 }
 
-/// Checks if the expressions matches
-/// ```rust
-/// { static __STATIC_FMTSTR: &[""] = _; __STATIC_FMTSTR }
-/// ```
-fn check_static_str(cx: &LateContext, expr: &Expr) -> bool {
+/// Returns the slice of format string parts in an `Arguments::new_v1` call.
+/// Public because it's shared with a lint in print.rs.
```
Comment (mcarton) [inline clippy_lints/src/format.rs]: Small nit: I had made a `higher` module for that kind of un-lowering/un-expanding stuffs.

File: tests/compile-fail/print_with_newline.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+#![deny(print_with_newline)]
+
+fn main() {
+    print!("Hello");
+    print!("Hello\n"); //~ERROR using `print!()` with a format string
+    print!("Hello {}\n", "world"); //~ERROR using `print!()` with a format string
+    print!("Hello {} {}\n\n", "world", "#2"); //~ERROR using `print!()` with a format string
```
Comment (mcarton) [inline tests/compile-fail/print_with_newline.rs]: If there are more that one `\n`, `print!` is a valid alternative IMO.

**Pull request merged**